source/Halibut.Tests/TraceLogFileLogger.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.IO;
+
+namespace Halibut.Tests
+{
+    public class TraceLogFileLogger : IDisposable
+    {
+        string tempFilePath = Path.GetTempFileName();
+        string testHash;
+
+        public void SetTestHash(string testHash)
+        {
+            this.testHash = testHash;
+        }
+        
+        public void WriteLine(string logMessage)
+        {
+            File.AppendAllLines(tempFilePath, new[] { logMessage });
+        }
+
+        public bool CopyLogFileToArtifacts()
+        {
+            var currentDirectory = Directory.GetCurrentDirectory();
+            var rootDirectory = new DirectoryInfo(currentDirectory).Parent.Parent.Parent.Parent.Parent;
+            var traceLogsDirectory = rootDirectory.CreateSubdirectory("artifacts").CreateSubdirectory("trace-logs");
+            var fileName = $"{testHash}.tracelog";
+
+            try
+            {
+                File.Copy(tempFilePath, Path.Combine(traceLogsDirectory.ToString(), fileName), true);
```

Comment:
Could we use 'TestContext.AddTestAttachment' here as well/instead?

Reply:
I did try that actually, and it worked well locally but TeamCity threw a wobbly i.e. exceptions in the JVM, so I removed it

When running in TeamCity, the file name for the attached file is something long and unfriendly (I think it might've been a URL-encoded string of the full test name, which gets _really_ long and hard to read.

Maybe I could attach the file to local tests only, so at least it's easier for local dev?

Reply:
All good. It sounds like this option was well explored (and found wanting 😁 )